systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/AbstractKafkaClient.java

```diff
@@ -142,48 +170,67 @@ public void verifyProducedAndConsumedMessages(int producedMessages, int consumed
      * Get external bootstrap connection
      * @param namespace kafka namespace
      * @param clusterName kafka cluster name
+     * @param listenerName name of the listener
      * @return bootstrap url as string
      */
-    public static String getExternalBootstrapConnect(String namespace, String clusterName) {
+    @SuppressWarnings("Regexp") // because of extBootstrapService.getSpec().getType().toLowerCase()
+    @SuppressFBWarnings("DM_CONVERT_CASE")
+    public static String getExternalBootstrapConnect(String namespace, String clusterName, String listenerName) {
```

Comment:
When you deploy Kafka, the bootstrap addresses for all listeners get filled in the status section. Is there any actual reason why you need to reverse engineer the address here instead of reading it simply from the status? All you would need is a listener name, you would not need to both with OpenShfit, service types etc.